### PR TITLE
[Fabric][Upgrade] Upgrade RN to 0.73.6 so we can patch Fabric component for RN>0.73

### PR DESCRIPTION
**Pull request by shwanton**
# Summary

NOTE: Breaking Change for RN 0.73!

- Fabric component descriptor api was changed in 0.73
- Since DateTimePicker uses a custom renderer for iOS/MacOS, we need to patch the Component Descriptor for anyone using RN>0.73
  - Patch from Expo: https://github.com/expo/expo/blob/main/tools/src/react-native-nightlies/patches/datetimepicker.patch 
- We first need to upgrade RN to RN 0.73 & RNTA to 3.5.3

### iOS - Paper
https://github.com/react-native-datetimepicker/datetimepicker/assets/96719/899a1cdf-f742-4029-87d8-d4938873cd62

### iOS - Fabric
https://github.com/react-native-datetimepicker/datetimepicker/assets/96719/da023375-8155-4409-8974-7eda848339ee

### What's required for testing (prerequisites)?

Run w/ Paper & New Arch on RN 0.73

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ✅     |
| Android |    ✅      |

## Checklist

- [x] I have tested this on a device and a simulator
- [ ] I added the documentation in `README.md`
- [ ] I updated the typed files (TS and Flow)
- [ ] I added a sample use of the API in the example project (`example/App.js`)
- [ ] I have added automated tests, either in JS or e2e tests, as applicable

### CODE REVIEW

**Comment by arushikesarwani94** [comment]
## New Arch Enabled - Android


https://github.com/react-native-datetimepicker/datetimepicker/assets/76862293/9bc646e9-03eb-4178-9f83-dcb765dabe18

File: example/ios/Podfile [hunk 1]
```diff
@@ -2,9 +2,4 @@ require_relative '../../node_modules/react-native-test-app/test_app'
 
 workspace 'date-time-picker-example.xcworkspace'
 
-# Flipper causes the build to fail on release when fabric is enabled
-# https://github.com/facebook/react-native/issues/33764
-
-use_flipper! false unless ENV['ENABLE_FLIPPER']
-
```
Comment (shwanton) [inline example/ios/Podfile:9]: `use_flipper` is not supported in 0.73

**Comment by shwanton** [comment]
I'm fixing the flow errors. Has there been any attempts to use Flow Multiplatform? https://flow.org/en/docs/react/multiplatform/

**Comment by vonovak** [comment]
> I'm fixing the flow errors. Has there been any attempts to use Flow Multiplatform? https://flow.org/en/docs/react/multiplatform/

hello, no, there haven't been any. The idea was to stop using Flow and use TS to be honest, though there's no timeline for that change.

**Comment by shwanton** [comment]
Closing this PR in favor of #879

**Pull request closed without merging**